frontend/src/metabase/dashboard/containers/CreateDashboardForm.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useCallback, useMemo } from "react";
+import { t } from "ttag";
+import _ from "underscore";
+import * as Yup from "yup";
+import { connect } from "react-redux";
+import { withRouter } from "react-router";
+
+import Button from "metabase/core/components/Button";
+import Form from "metabase/core/components/Form";
+import FormFooter from "metabase/core/components/FormFooter";
+import FormProvider from "metabase/core/components/FormProvider";
+import FormInput from "metabase/core/components/FormInput";
+import FormTextArea from "metabase/core/components/FormTextArea";
+import FormSubmitButton from "metabase/core/components/FormSubmitButton";
+import FormErrorMessage from "metabase/core/components/FormErrorMessage";
+import FormCollectionPicker from "metabase/core/components/FormCollectionPicker";
+
+import * as Errors from "metabase/core/utils/errors";
+
+import Collections from "metabase/entities/collections";
+import Dashboards from "metabase/entities/dashboards";
+
+import type { CollectionId, Dashboard } from "metabase-types/api";
+import type { State } from "metabase-types/store";
+
+const DASHBOARD_SCHEMA = Yup.object({
+  name: Yup.string().required(Errors.required).max(100, Errors.maxLength),
+  description: Yup.string().nullable().max(255, Errors.maxLength),
+  collection_id: Yup.number().nullable(),
+});
+
+interface CreateDashboardProperties {
+  name: string;
+  description: string | null;
+  collection_id: CollectionId;
+}
+
+export interface CreateDashboardFormOwnProps {
+  collectionId?: CollectionId; // can be used by `getInitialCollectionId`
+  onCreate?: (dashboard: Dashboard) => void;
+  onCancel?: () => void;
+}
+
+interface CreateDashboardFormStateProps {
+  initialCollectionId: CollectionId;
+}
+
+interface CreateDashboardFormDispatchProps {
+  handleCreateDashboard: (
+    dashboard: CreateDashboardProperties,
+  ) => Promise<Dashboard>;
+}
+
+type Props = CreateDashboardFormOwnProps &
+  CreateDashboardFormStateProps &
+  CreateDashboardFormDispatchProps;
+
+function mapStateToProps(state: State, props: CreateDashboardFormOwnProps) {
+  return {
+    initialCollectionId: Collections.selectors.getInitialCollectionId(
+      state,
+      props,
+    ),
+  };
+}
+
+const mapDispatchToProps = {
+  handleCreateDashboard: Dashboards.actions.create,
+};
+
+function CreateDashboardForm({
+  initialCollectionId,
+  handleCreateDashboard,
+  onCreate,
+  onCancel,
+}: Props) {
+  const initialValues = useMemo(
```

Comment:
You can generate initial values from schema, example https://github.com/metabase/metabase/blob/629e9cfebbbb787eed25cdf9fff3b03dba643b59/frontend/src/metabase/account/password/components/UserPasswordForm/UserPasswordForm.tsx#L41

Reply:
You'd need to add `default()` to every field in the schema for that

Reply:
Nice, I didn't know about it, thanks! This would definitely work for `name` and `description`, but the default `collection_id` value is coming from the `getInitialCollectionId` selector. We can combine it like this:

```js
const initialValues = useMemo(() => ({
   ...DASHBOARD_SCHEMA.getDefault(),
   collection_id: initialCollectionId,
}, [initialCollectionId]);
```

Does that look good, or is there a better way to achieve the same thing?

Reply:
No I think that's the way to do it afaik